fixes #4201 - update operating system by label

isratrade opened the pull request:
operatingsystem_names uses `to_label` which returns description if it exists or fullname (name, major, minor)

domcleal commented on app/models/operatingsystem.rb:
```diff
@@ -131,8 +132,9 @@ def fullname
-  def self.find_by_fullname(fullname)
-    a = fullname.split(" ")
+  def self.find_by_to_label(str)
+    return @os if (@os = Operatingsystem.find_by_description(str))
```
nitpick, no need for an instance variable here?

domcleal replied:
s/Operatingsystem/self/ ?

domcleal commented:
I've tried doing a `GET /api/v2/operatingsystems/foo` (where an OS has its description set to `foo`) but the OS isn't found.  SQL logs show it's trying to search by name and not finding it.

Do we need to define resource_identifying_attributes with `to_label` instead for it to work?

Also I think routes might need changing to accept names containing full stops, e.g. `CentOS 6.5` which comes through to the controller as `CentOS 6`.

isratrade commented on app/models/operatingsystem.rb:
```diff
@@ -119,6 +120,11 @@ def to_label
 
+  # to_label setter updates description and does not try to parse and update major, minor attributes
+  def to_label=(str)
+    self.description = str
```
@domcleal, added this setter method and test and re-committed.

isratrade commented:
@domcleal, recommited
- edited routes to allow id with spaces or dots.  api/v2/operatingsystems/RHEL%206.3
- added `to_label` to resource_identifying_attributes
- added some tests

isratrade commented on app/models/operatingsystem.rb:
```diff
@@ -116,23 +117,31 @@ def interpolate_medium_vars path, arch, os
-    a = fullname.split(" ")
+  def self.find_by_to_label(str)
+    os = self.find_by_description(str)
+    return os if os
```
not sure why this one-liner didn't work

``` ruby
    return os if (os = self.find_by_description(str))
```

I get the error

```
NameError: undefined local variable or method `os' for #<Class:0x0000000afb4a78>
```

so I split it into two lines and it works ????

domcleal commented:
@isratrade please rebase

Commit:
fixes #4201 - update operating system by label

operatingsystem_names uses to_label which returns description if it exists or fullname (name, major, minor) if description is blank

isratrade commented:
rebased and re-commited

domcleal commented:
Merged as 42c9541 for Foreman 1.5.0, thanks @isratrade!
